File: src/turtlebot_controller.cpp
#include "ros/ros.h"
#include "turtlesim/Pose.h"
#include "geometry_msgs/Twist.h"
#include "turtlesim/Spawn.h"

void turtleCallback(const turtlesim::Pose::ConstPtr& msg)
{
ROS_INFO("Turtle subscriber@[%f, %f, %f]",
msg->x, msg->y, msg->theta);
}

int main (int argc, char **argv)
{
	ros::init(argc, argv, "turtlebot_subscriber");
	ros::NodeHandle nh;
	ros::Subscriber sub = nh.subscribe("my_new_turtle/pose", 1,turtleCallback); 
	ros::Publisher pub = nh.advertise<geometry_msgs::Twist>("my_new_turtle/cmd_vel", 1);
	ros::ServiceClient client = nh.serviceClient<turtlesim::Spawn>("/spawn");
	
	turtlesim::Spawn spawn;
	spawn.request.x = 1.0;
	spawn.request.y = 1.0;
	spawn.request.theta = 0.0;
	spawn.request.name = "my_new_turtle";
	
	client.call(spawn);
	
	while (ros::ok()){
		geometry_msgs::Twist my_vel;
		my_vel.linear.x = 0.1;
		pub.publish(my_vel);
		ros::spinOnce();
	}
	return 0;
}

File: scripts/turtlebot_controller.py
#!/usr/bin/env python
import rospy
from turtlesim.msg import Pose
from geometry_msgs.msg import Twist

def callback_position(pose):
    rospy.loginfo("The robot is in %f, %f, %f", pose.x, pose.y, pose.theta)
    
def controller():

    rospy.init_node('controller', anonymous=True)
    rospy.Subscriber("turtle1/pose", Pose, callback_position)
    pub = rospy.Publisher('turtle1/cmd_vel', Twist, queue_size=10)
    while not rospy.is_shutdown():
        vel = Twist()
        vel.linear.x = 0.1
        pub.publish(vel)

if __name__ == '__main__':
    controller()
